stream_alert/apps/_apps/intercom.py

```diff
@@ -0,0 +1,103 @@
+import time
+import re
+
+import requests
+
+from . import AppIntegration, StreamAlertApp, get_logger
+
+
+LOGGER = get_logger(__name__)
+
+
+@StreamAlertApp
+class IntercomApp(AppIntegration):
+    """Intercom StreamAlert app"""
+    _INTERCOM_LOGS_URL = 'https://api.intercom.io/admins/activity_logs'
+
+    def __init__(self, event, config):
+        super(IntercomApp, self).__init__(event, config)
+        self._next_page = None
+
+    @classmethod
+    def _type(cls):
+        return 'admin_activity_logs'
+
+    @classmethod
+    def service(cls):
+        return 'intercom'
+
+    @classmethod
+    def _required_auth_info(cls):
+        return {
+            'token': {
+                'description': 'the OAuth token for this Intercom app',
+                'format': re.compile(r'.*')
+            }
+        }
+
+    def _sleep_seconds(self):
+        """Return the number of seconds this polling function should sleep for
+        between requests to avoid failed requests. Intercom API allows for a default of 500 requests
+        every minute, distributed over 10 seconds periods. This means for a default rate limit of
+        500 per minute, you can send a maximum of 83 operations per 10 second period.
+
+        Resource(s):
+            https://developers.intercom.com/intercom-api-reference/reference#rate-limiting
+
+        Returns:
+            int: Number of seconds that this function should sleep for between requests
+        """
+        return 10
+
+    def _gather_logs(self):
+        # Generate headers
+        headers = {'Authorization': "Bearer %s" % self._config.auth['token'],
+                   'Accept': 'application/json'}
+
+        # Results are paginated with a page url field provided that is used in subsequent queries.
+        # If this field exists, make a a query to the page url, and if not, make a fresh query to
+        # the default _INTERCOM_LOGS_URL with the required created_at_before and created_at_after
+        # parameters
+        if self._next_page:
+            params = None
+            url = self._next_page
+        else:
+            params = {'created_at_before': int(time.time()),
```

Comment:
is the `created_at_before` value in UTC? if so, it might be advisable to ensure you're always getting time in UTC (`time.time()` does local time and lambda runs in UTC, but enforcing UTC is usually safe when testing locally).

This seems clunky, but is the only way I know to get a consistent UTC timestamp in any locale in python 2:
```
import calendar
import time

current_time = int(calendar.timegm(time.gmtime()))
```

Reply:
thanks for the suggestion!